core_impl/src/test/java/com/google/instrumentation/trace/TraceExporterImplTest.java

```diff
@@ -141,6 +159,20 @@ public void exportSpansToMultipleServices() {
     assertThat(exported2.get(1)).isEqualTo(span2.toSpanData());
   }
 
+  @Test
+  public void exportNotSampledSpans() {
+    SpanImpl span1 = createNotSampledEndedSpan(SPAN_NAME_1);
+    SpanImpl span2 = createSampledEndedSpan(SPAN_NAME_2);
+    // Spans are recorded and exported in the same order as they are ended, we test that a non
+    // sampled span is not exported by creating and ending a sampled span after a non sampled span
+    // and checking that the first exported span is the sampled span (the non sampled did not get
+    // exported).
+    List<SpanData> exported = serviceHandler.waitForExport(1);
+    assertThat(exported.size()).isEqualTo(1);
+    assertThat(exported.get(0)).isNotEqualTo(span1.toSpanData());
```

Comment:
What if you created the spans separately from ending them, to make the export more explicit?

```
    SpanImpl span1 = createNotSampledSpan(SPAN_NAME_1);
    SpanImpl span2 = createSampledSpan(SPAN_NAME_2);
    span1.end();
    span2.end();
```

createSampledEndedSpan could call createSampledSpan.  It's also fine as it is.